### PR TITLE
Add Expanded CMS Support

### PR DESCRIPTION
**Pull request by davidzech**
* Support BER Encoded CMS
* Support Attached CMS
* Support Signed Attributes CMS

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -76,7 +76,7 @@ let package = Package(
 if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
     package.dependencies += [
         .package(url: "https://github.com/apple/swift-crypto.git", "2.5.0" ..< "4.0.0"),
-        .package(url: "https://github.com/apple/swift-asn1.git", from: "1.0.0"),
+        .package(url: "https://github.com/apple/swift-asn1.git", revision: "c7e239b5c1492ffc3ebd7fbcc7a92548ce4e78f0"),
```
Comment (davidzech) [inline Package.swift:79]: Waiting on a version bump from swift-asn1 to consume the new BER feature set.

**Comment by davidzech** [comment]
Leaving this in draft until [apple/swift-asn1 ](http://github.com/apple/swift-asn1) is tagged with a new version with BER support.

**Comment by dnadoba** [comment]
@davidzech we have just released [Swift ASN.1 1.1.0](https://github.com/apple/swift-asn1/releases/tag/1.1.0)

File: Package.swift (hunk 1 above)
Comment (davidzech) [reply Package.swift]: Fixed in 8713872df9c0bb57d6d6c1cc83cdec5a10399aa2

**Comment by davidzech** [comment]
> @davidzech we have just released [Swift ASN.1 1.1.0](https://github.com/apple/swift-asn1/releases/tag/1.1.0)

Ack. Updated and removed Draft status @dnadoba

**Comment by Lukasa** [comment]
@swift-server-bot add to allowlist

**Comment by Lukasa** [comment]
Based on CI, you need to add this block to your new files that import `Foundation`:

```
#if canImport(Darwin)
import Foundation
#else
@preconcurrency import Foundation
#endif
```

**Comment by davidzech** [comment]
> Based on CI, you need to add this block to your new files that import `Foundation`:
> 
> ```
> #if canImport(Darwin)
> import Foundation
> #else
> @preconcurrency import Foundation
> #endif
> ```

Ack. I likely need to update CMake again too.

**Review by Lukasa — APPROVED** [review]
API breakage is acceptable here.

**Pull request merged**